python/TestHarness/schedulers/Scheduler.py

```diff
@@ -533,9 +555,12 @@ def runWorker(self, job_container):
                 # Get next job list
                 next_job_group = self.getNextJobGroup(job_dag)
 
+                # Allow derived schedulers to perform post run operations
+                self.postRun(job_container)
```

Comment:
This should run _after_ getNextJobGroup, because getNextJobGroup deletes the current job.
Why; QueueManager checks for job_container size, and if its _zero_ it knows it can perform `qrls` on this group of jobs (PBS release all jobs in this group from hold). This allows us to release groups, instead of the traditional release of all (thousands) of jobs at once, at the end.